### PR TITLE
[terra-clinical-data-grid] Allow hover styles to apply only on non-touch devices

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
This PR resolves the issue by making use of hover and pointer interaction media features introduced in Media Queries Level 4 so that the hover styles are applied only on non-touch devices whose primary input is a fine pointer device such as a mouse or a stylus

### Deployment Link

https://terra-clinical-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by emilyrohrbough** [comment]
@saket2403 why did the screenshots change with this media query? Are we being too specific with `and (pointer: fine)` included in the query?

**Comment by StephenEsser** [comment]
@emilyrohrbough This may be a bug with headless-chrome. Running in non-headless Chrome the screenshots are generated as expected with the media queries.  The styles also appear to be working correctly when running the site locally. I tried a few variations of the media query and was unsuccessful getting the screenshots to generate correctly.

Similar conversation thread here:

https://github.com/cerner/terra-core/pull/2929#issuecomment-620058376

**Comment by emilyrohrbough** [comment]
@StephenEsser Thanks for the info.

**Comment by saket2403** [comment]
Changes reverted to use media queries and wdio tests removed for hover styles.

File: packages/terra-clinical-data-grid/src/subcomponents/Cell.module.scss [hunk 1]
```diff
@@ -39,9 +41,12 @@
         box-shadow: var(--terra-clinical-data-grid-cell-selected-focus-box-shadow, inset 0 0 0 1px #0092e0, inset 0 0 0 4px rgba(76, 178, 233, 0.3));
         outline: none;
       }
-
-      &:hover {
-        background-color: var(--terra-clinical-data-grid-cell-selected-hover-background-color, #b3ddf7);
+      
+      /* stylelint-disable max-nesting-depth */
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/subcomponents/Cell.module.scss:45]: can we disable next line or does this need to be re-enabled?

File: packages/terra-clinical-data-grid/src/subcomponents/Scrollbar.module.scss [hunk 2]
```diff
@@ -17,11 +17,17 @@
     display: none;
   }
 
-  .container:hover,
+
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/subcomponents/Scrollbar.module.scss:20]: extra line

Comment (saket2403) [reply packages/terra-clinical-data-grid/src/subcomponents/Scrollbar.module.scss]: Removed 9225f6d

File: packages/terra-clinical-data-grid/src/subcomponents/Cell.module.scss (hunk 1 above)
Comment (saket2403) [reply packages/terra-clinical-data-grid/src/subcomponents/Cell.module.scss:45]: Re-enabled 9225f6d

**Comment by neilpfeiffer** [comment]
Tested: PR #652
Resolves: [Terra-Core: Issue #2927](https://github.com/cerner/terra-core/issues/2927)

**Reviewed:**
Manually reviewed the hover states for the components affected:
- Clinical-data-grid

\
\
Verified by @neilpfeiffer, ready to merge after engineering review.

**Pass** |
-----------|
UX Review|

**Comment by ryanthemanuel** [comment]
Spoke to Paul and he was good with just UX review on this.

**Pull request merged**